zapcore/error.go

```diff
@@ -75,9 +77,41 @@ func encodeError(key string, err error, enc ObjectEncoder) (retErr error) {
 			enc.AddString(key+"Verbose", verbose)
 		}
 	}
+
+	// Unwrap the error and marshal any ObjectMarshalers in the error chain.
+
+	marshalers := make(mergedMarshalers, 0, 1)
```

Comment:
out of curiosity, why pre-allocate? I thought creating a nil slice `var marshalers mergedMarshalers` would be just as good for single error (we allocate on the heap once?) but better if the error doesn't implement the `ObjectMarhaller` which is typically the case?